src/Donations/Controllers/DonationRequestController.php

```diff
@@ -0,0 +1,210 @@
+            );
+        }
+
+        $isAdmin = current_user_can('manage_options');
+
+        $includeSensitiveData = $request->get_param('includeSensitiveData');
+        if ( ! $isAdmin && $includeSensitiveData) {
+            return new WP_REST_Response(
+                ['message' => __('You do not have permission to include sensitive data.', 'give')],
+                403
+            );
+        }
+
+        $donationAnonymousMode = new DonationAnonymousMode($request->get_param('anonymousDonations'));
+        if ( ! $isAdmin && $donation->anonymous && ! $donationAnonymousMode->isRedacted()) {
+            return new WP_REST_Response(
+                ['message' => __('You do not have permission to include anonymous donations.', 'give')],
+                403
+            );
+        }
+
+        return new WP_REST_Response($this->escDonation($donation, $includeSensitiveData, $donationAnonymousMode));
+    }
+
+    /**
+     * @unreleased
+     */
+    public function getDonations(WP_REST_Request $request): WP_REST_Response
+    {
+        $isAdmin = current_user_can('manage_options');
+
+        $includeSensitiveData = $request->get_param('includeSensitiveData');
+        if ( ! $isAdmin && $includeSensitiveData) {
+            return new WP_REST_Response(
+                ['message' => __('You do not have permission to include sensitive data.', 'give')],
+                403
+            );
+        }
+
+        $donationAnonymousMode = new DonationAnonymousMode($request->get_param('anonymousDonations'));
+        if ( ! $isAdmin && $donationAnonymousMode->isIncluded()) {
+            return new WP_REST_Response(
+                ['message' => __('You do not have permission to include anonymous donations.', 'give')],
+                403
+            );
+        }
+
+        $page = $request->get_param('page');
+        $perPage = $request->get_param('per_page');
+        $sortColumn = $this->getSortColumn($request->get_param('sort'));
+        $sortDirection = $request->get_param('direction');
+        $mode = $request->get_param('mode');
+
+        $query = Donation::query();
+
+        if ($campaignId = $request->get_param('campaignId')) {
+            // Filter by CampaignId
+            $query->where('give_donationmeta_attach_meta_campaignId.meta_value', $campaignId);
+        }
+
+        if ($donationAnonymousMode->isExcluded()) {
+            // Exclude anonymous donations from results
+            $query->where('give_donationmeta_attach_meta_anonymous.meta_value', 0);
+        }
+
+        // Include only current payment "mode"
+        $query->where('give_donationmeta_attach_meta_mode.meta_value', $mode);
+
+        // Include only valid statuses
+        $query->whereIn('post_status', ['publish', 'give_subscription']);
+
+        $query
+            ->limit($perPage)
+            ->offset(($page - 1) * $perPage)
+            ->orderBy($sortColumn, $sortDirection);
+
+        $donations = $query->getAll() ?? [];
```

Comment:
One thing I just noticed, is we are directly returning the query results.  The Model query builder returns an array of models so not all the properties will be readable.  So we actually will need to map over the results and call `$donation->toArray()`.

Reply:
oh wait is that what escDonation is doing?  maybe there's just a few properties we need to adjust for.

Reply:
@jonwaldstein Exactly! the `escDonation()` method is doing that.